### DISCUSSION TITLE
API Post Zone with UTF-8 Characters like ü,ä

### DISCUSSION THREAD

**Discussion by Kr0san89** [PowerDNS/pdns#13398] [Q&A]
Hello,

i tried to add a new Zone on Endpoint POST /api/v1/servers/localhost/zones with a body
`{"name": "büro.de.", "kind": "Native", "masters": [], "nameservers": ["ns1.example.org.", "ns2.example.org."]}`

But the Server returns (HTTP Status 422 Unprocessable Entity)
`{
  "error": "Name 'b\\195\\188ro.de.' contains unsupported characters"
}`

further i tried to add it with the encoding inside: (same result)
`{"name": "b\\195\\188ro.de.", "kind": "Native", "masters": [], "nameservers": ["ns1.example.org.", "ns2.example.org."]}`

The interesting thing is you can create such zones with 
`pdnsutil create-zone``
and you can even read them with the Endpoint GET /api/v1/servers/localhost/zones/büro.de.

**I was wondering if there is a way to create zones with the API including UTF-8 characters.**

I also found this issue with some similar issue but on another Endpoint https://github.com/PowerDNS/pdns/issues/12725 
Does the stricter validation also apply here? and why is it even stricter?

**Comment by Habbie** [comment]
If you type a unicode name into your browser, it will not go out and do UTF-8 queries, it will use Punycode. Why do you want to put UTF-8 in your database?

**Reply by Kr0san89** [reply]
i need a way to get ü,ä and similar characters into powerdns as Zone over the API

**Reply by Habbie** [reply]
For what purpose?

**Reply by Kr0san89** [reply]
having ü.ä and so in is valid inside domain names and actually used in german language space. Otherwise i can not create them via API?

**Reply by Habbie** [reply]
https://en.wikipedia.org/wiki/Internationalized_domain_name#Example_of_IDNA_encoding

**Reply by Kr0san89** [reply]
thx now i understand you first answer :)
